### PR TITLE
Compute executionOptimistic for state/events routes

### LINKED ISSUE #4495: Implement executionOptimistic flag in API impl
- PR https://github.com/ChainSafe/lodestar/pull/4493 filled executionOptimistic in all methods that required it, but the implementation is missing for some

Compute correct executionOptimistic flag for all remaining API routes. Remove usage of this constant

https://github.com/ChainSafe/lodestar/blob/6a37cca422c5e6b819d123e0dd456ee994e08346/packages/beacon-node/src/api/impl/types.ts#L10-L15

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

- A continuation of https://github.com/ChainSafe/lodestar/pull/4943/files
- Follow beacon api spec

**Description**

- Implement `execution_optimistic` for state and events api
- Remove `IS_OPTIMISTIC_TEMP` complete

Closes #4495

### CODE REVIEW

File: packages/beacon-node/src/api/impl/events/index.ts [hunk 1]
```diff
@@ -33,35 +35,42 @@ export function getEventsApi({chain, config}: Pick<ApiModules, "chain" | "config
...
-        newHeadBlock: newHead.blockRoot,
-        oldHeadBlock: oldHead.blockRoot,
-        newHeadState: newHead.stateRoot,
-        oldHeadState: oldHead.stateRoot,
-        executionOptimistic: IS_OPTIMISTIC_TEMP,
+        executionOptimistic: false,
       },
     ],
+    [routes.events.EventType.chainReorg]: (oldHead, newHead, depth) => {
+      const blockSummary = forkchoice.getBlock(fromHexString(newHead.blockRoot));
```
Comment (dapplion) [inline packages/beacon-node/src/api/impl/events/index.ts]: @tuyennhv if you change the type of Chain.reorg to match EventType.chainReorg, you won't need to do this conversion, and won't have to do forkchoice calls in this event transformation.

File: packages/beacon-node/src/api/impl/events/index.ts [hunk 2]
```diff
@@ -33,35 +35,42 @@ export function getEventsApi({chain, config}: Pick<ApiModules, "chain" | "config
...
   } = {
     [routes.events.EventType.head]: (data) => [data],
-    [routes.events.EventType.block]: (block) => [
-      {
-        block: toHexString(config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message)),
-        slot: block.message.slot,
-        executionOptimistic: IS_OPTIMISTIC_TEMP,
-      },
-    ],
+    [routes.events.EventType.block]: (block) => {
```
Comment (dapplion) [inline packages/beacon-node/src/api/impl/events/index.ts]: Same as below, change the ChainEvent.block type to match EventType.block so you don't need to re-compute the root here, nor do forkchoice calls

File: packages/beacon-node/src/api/impl/events/index.ts [hunk 3]
```diff
@@ -24,6 +25,7 @@ const chainEventMap = {
 };
 
 export function getEventsApi({chain, config}: Pick<ApiModules, "chain" | "config">): routes.events.Api {
+  const forkchoice = chain.forkChoice;
```
Comment (dapplion) [inline packages/beacon-node/src/api/impl/events/index.ts]: Should not do forkchoice calls here

File: packages/beacon-node/test/unit/api/impl/events/events.test.ts [hunk 4]
```diff
@@ -24,6 +25,7 @@ describe("Events api impl", function () {
       chainStub.stateCache = (stateCacheStub as unknown) as StateContextCache;
       chainEventEmmitter = new ChainEventEmitter();
       chainStub.emitter = chainEventEmmitter;
+      (chainStub as {forkChoice: IForkChoice}).forkChoice = ({getBlock: sinon.stub()} as unknown) as IForkChoice;
```
Comment (dapplion) [inline packages/beacon-node/test/unit/api/impl/events/events.test.ts]: This stub is not necessary anymore right?

Comment (twoeths) [reply packages/beacon-node/test/unit/api/impl/events/events.test.ts]: no it's not necessary anymore, I removed it 👍

**Review by dapplion — APPROVED** [review]

**Comment by dapplion** [comment]
Sim tests failed twice in a row after retrying

**Comment by twoeths** [comment]
> Sim tests failed twice in a row after retrying

@dapplion  that's common error and not specific to this PR, I created https://github.com/ChainSafe/lodestar/issues/4960 for a separate investigation

**Pull request merged**